### PR TITLE
Add providers, operators, and hooks readmes

### PR DESCRIPTION
**Pull request by petedejoy**
It took me quite a while to piece together how exactly operators, hooks, and providers would be released and distributed in Airflow 2.0 and beyond from various mailing list threads, AIPs, and slack conversations. Hoping that adding a little bit of color to readmes in the relevant subdirectories will help users who may be less familiar with the maintenance of the project figure out how exactly the operators, hooks, and providers directories differ. Someone should give these a quick once-over to validate that my mental model is accurate.

---

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
NICE!

**Comment by potiuk** [comment]
Just one EOF static check failiing :)

**Comment by petedejoy** [comment]
Thanks @potiuk! Is that failing because I need to add a new line at the end of these files?

File: airflow/operators/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+ -->
+
+# Airflow Operators
+
+Airflow operators are modules that represent a single, ideally idempotent task. They are arbiters of the logic that executes when your DAG runs.
```
Comment (ryw) [inline airflow/operators/README.md:22]: I'd capitalize "Airflow Operators"

**Review by ryw — APPROVED** [review]

**Pull request merged**